File: backend/src/main/java/dev/flima/backend/domain/model/Product.java
package dev.flima.backend.domain.model;

import lombok.Data;
import java.math.BigDecimal;
import java.util.List;

@Data
public class Product {

    private Long id;
    private String code;
    private String name;
    private BigDecimal price;
    private List<ProductMaterial> materials;

}

File: frontend/src/types/Product.ts
export interface ProductMaterial {
  id: number;
  rawMaterialId: number;
  rawMaterialName: string;
  requiredQuantity: number;
}

export interface Product {
  id: number;
  code: string;
  name: string;
  price: string;
  materials?: ProductMaterial[];
}